### DISCUSSION TITLE
script to get few device's inventory via form

### DISCUSSION THREAD

**Discussion by nikunj-tele** [netbox-community/netbox#15467] [Help Wanted!]
Hi experts,

I want  to write the netbox script to export/view the entered the device name or serial number , I wrote first part where it asks for the device name/serial number but getting issue with second part so it would be kind if any one can help with the further logic

```
import csv
import io
from dcim.models import Device, DeviceType, Manufacturer, Interface
from extras.models import Tag
from extras.scripts import Script, TextVar
from datetime import datetime
import re


    class BatchDeviceReport(Script):
    class Meta:
        name = "Report to get report of specific devices"
        description = "enter hostname one name per line"
    devices = TextVar(
        model=Device,
        label="Devices",
        description="Devices (Names, Serials or Asset tags) separated with newline",
    )
   
    def run(self, data, commit):
        output = io.StringIO()
        writer = csv.writer(output)
        print ( )
       
       for device in data["devices"]:
            dt = device.name
            print(device.name,serial.number)
```
I tried following code but it second part did not work  , I am struglling with data["devices"]: 
in nbshell following is working but in my script "devices " entered in form (assigned to data variable) , how can I poss the variable in the script which can be printed and exported to csv.
```
>>> for device in Device.objects.all():
...   print(device.name)
...
device01
device012
```

TIA

**Comment by candlerb** [comment]
Your formatting is mangled: `class BatchDeviceReport(Script):` needs to align with the left-hand side.

If you want users to select a set of devices from drop-down menus then you want MultiObjectVar.

If you want them to enter plain text, then TextVar is fine (but it will not have `model` as an attribute - see [docs](https://docs.netbox.dev/en/stable/customization/custom-scripts/#variable-reference)), and then it's up to you to parse the text, split it into lines and then perform Django ORM queries to search for objects. At this point, it's a question of Python/Django programming; you might want to ask further questions on a Django mailing list, or have a look at the Netbox source code for how it handles `q` (search) queries.

**Reply by nikunj-tele** [reply]
Thanks @candlerb ,

I have fixed the Django part and I can see the form to get the data !

<img width="755" alt="image" src="https://github.com/netbox-community/netbox/assets/55275995/c42600e6-6ec5-4aab-8ae9-654568c16647">

**Reply by nikunj-tele** [reply]
now how can use the data passed(variable) the in the form  and use this variable in netbox to get the report. it might be basic question but If any reference/pointer is there then would be great help
Thanks
